File: test/timing/hessian/hessian_speed.jl
# Add paths
using DSGE, MATLAB
path = dirname(@__FILE__)

# Load variables
mf = MatFile("$path/hessian.mat")
params = get_variable(mf, "params")
YY0 = get_variable(mf, "YY0")
YY = get_variable(mf, "YY")
close(mf)
model = Model990()
data = [YY0; YY]

# Call hessian
tic()
hessizero!(params, model, data; noisy=false)
time_elapsed = toq()

for node in ARGS
    println("node: $node, seconds: $time_elapsed")
end

File: test/timing/hessian/hessian_speed.m
function [] = hessian_speed(node)

% Add paths
path = '/data/dsge_data_dir/proc/dsge/pearl/cleanCode990/';
dirs = {[path, 'data/']; [path, 'dsgesolv/']; [path, 'estimation/']; [path, 'forecast/']; ...
        [path, 'initialization/']; [path, 'kalman/']; [path, 'toolbox/']; [path, 'plotting/']};
addpath(dirs{:});

% Load variables
load('hessian.mat');

tic;
hessizero('objfcndsge', [params, para_mask], 0, ...
    YY, YY0, nobs, nlags, nvar, mspec, npara, trspec, pmean, pstdd, pshape, para_mask, ...
    para_fix, marglh, coint, cointadd, cointall, YYcoint0, MIN, args_nant_antlags{:});
time_elapsed = toc;

disp(['node: ', node, ' seconds: ',  num2str(time_elapsed)]);
exit;

end